blas/testblas/common.go

```diff
@@ -254,13 +254,46 @@ func allPairs(x, y []int) [][2]int {
 	return p
 }
 
+func sameFloat64(a, b float64) bool {
+	return a == b || math.IsNaN(a) && math.IsNaN(b)
+}
+
+func sameComplex128(x, y complex128) bool {
+	return sameFloat64(real(x), real(y)) && sameFloat64(imag(x), imag(y))
+}
+
 func zsame(x, y []complex128) bool {
 	if len(x) != len(y) {
 		return false
 	}
 	for i, v := range x {
 		w := y[i]
-		if v != w && !math.IsNaN(real(v)) && !math.IsNaN(imag(v)) && !math.IsNaN(real(w)) && !math.IsNaN(imag(w)) {
+		if !sameComplex128(v, w) {
+			return false
+		}
+	}
+	return true
+}
+
+// zEqualApprox returns whether vectors x and y with stride inc
```

Comment:
Thanks. I'm not convinced of the name, but I can't think of anything better and it doesn't face the public, so we'll leave it.

Reply:
What about the name worries you? I'll merge but I think there may be a chance to revisit this later. In general, the supporting functions for blas testing could be cleaned up and unified.

Reply:
The name is indicating that it's testing for approximate equality, but there is no tolerance that will allow NaN == NaN, but conversely, those elements are not the elements that are being tested for approximate equality. A name that does cover it is `zApproxEqualInElementsSameOtherwise`; this is obviously not a good name though.

Reply:
I think that having two functions, one for stride elements and one for out-of-stride elements, would be better. I'll try to do something about it.